thunder/core/proxies.py

```diff
@@ -1238,10 +1238,14 @@ def _infer_tensor_properties(
 
     # dynamic shape not yet enabled, otherwise, the bake in should be guarded with if not using_symbolic_values():
     # dynamic shape support is currently block by #471 https://github.com/Lightning-AI/lightning-thunder/issues/471
-    _shape = tuple(pyval(x) for x in _shape)
-
-    # Computes derived properties
-    _numel = reduce(operator.mul, _shape, 1)
+    if not using_symbolic_values():
+        _shape = tuple(pyval(x) for x in _shape)
+        # Computes derived properties
+        _numel = reduce(operator.mul, _shape, 1)
+    else:
+        # deferred computation of numel
```

Comment:
Why defer the numel computation?

Reply:
two issues:
1. Since this function is executed/traced before the TensorProxy is constructed. We'll be accessing the `shape` attributes. So deferring this naturally resolves that.
2. numel computation may or may not be used at all. So if we defer the use, we don't have to compute it in the trace.

Note that we'll be relying on CSE to remove duplicated `numel` computation across the board.

Reply:
This seems OK for now, but wouldn't precomputing numel just give another number proxy if the shape contains one or more number proxies? Then requests for numel would just return that proxy? That seems like nbd to compute at compiletime and not have to compute as part of the computation?

Reply:
That's true. It's just some extra things in the trace that could also be removed with DCE or something.

I can have a follow up PR try to clean this up a bit. I just need to figure out how/where to place the computation of numel in trace.